core/src/worker/workflow/mod.rs

```diff
@@ -489,6 +488,22 @@ impl Workflows {
         }
         reserved
     }
+
+    /// Wraps responding to legacy queries. Handles ignore-able failures.
+    async fn respond_legacy_query(
+        &self,
+        tt: TaskToken,
+        res: QueryResult,
+    ) -> Result<(), tonic::Status> {
+        match self.client.respond_legacy_query(tt, res).await {
+            Ok(_) => Ok(()),
+            Err(e) if e.code() == tonic::Code::NotFound => {
+                warn!(error=?e,"Query not found when attempting to respond to it");
```

Comment:
I'm assuming you want to warn to let users know that query handling was too slow.. if that's the rationale LGTM

Reply:
It is :)